src/bindings/python/tests/test_runtime/test_compiled_model.py

```diff
@@ -27,6 +29,9 @@ def test_get_runtime_model(device):
     compiled_model = generate_relu_compiled_model(device)
     runtime_model = compiled_model.get_runtime_model()
     assert isinstance(runtime_model, Model)
+    # make sure runtime_model is destroyed before the dynamic library is unloaded, as described in issue #18388
+    del runtime_model
+    gc.collect()
```

Comment:
we have the following code:
https://github.com/openvinotoolkit/openvino/blob/8884e55b63a7c50f58f6137e9d3140813ed626f9/src/frontends/common/src/frontend.cpp#L19-L30
which adds a reference to frontend library to `ov::Model` when it's read by frontends.

In current case, `runtime_model` is `ov::Model` created by plugins. Do you think we can fix the issue similarly to frontends case somewhere here?
https://github.com/openvinotoolkit/openvino/blob/8884e55b63a7c50f58f6137e9d3140813ed626f9/src/inference/src/cpp/compiled_model.cpp#L34-L36
For this we need to add a `CompiledModel` as a `friend` similarly to:
https://github.com/openvinotoolkit/openvino/blob/8884e55b63a7c50f58f6137e9d3140813ed626f9/src/core/include/openvino/core/model.hpp#L33-L44

Reply:
Hi, thanks for the review. I hadn't thought of solving this problem by recreating `ov::Model` in the main runtime. I've created a new PR #22900. Could you please take a look at it?